tests/kernels/conftest.py

```diff
@@ -25,7 +27,20 @@ def create_kv_caches(
         key_cache = torch.empty(size=key_cache_shape,
                                 dtype=dtype,
                                 device=device)
-        key_cache.uniform_(-scale, scale)
+        if dtype != torch.uint8:
+            key_cache.uniform_(-scale, scale)
+        else:
+            # NOTE(zhaoyang): Due to NaN and Inf representation for fp8 data type,
+            # it may occur Inf or NaN if we directly use torch.randint
+            # to generate random data for fp8 cache.
+            # For example, s.11111.00 in fp8e5m2 format repesents Inf.
+            #     | E4M3        | E5M2
+            #-----|-------------|-------------------
+            # Inf | N/A         | s.11111.00
+            # NaN | s.1111.111  | s.11111.{01,10,11}
+            key_cache_tmp = torch.empty_like(key_cache, dtype=torch.float16)
+            key_cache_tmp.uniform_(-scale, scale)
+            cache_ops.convert_fp8(key_cache_tmp, key_cache)
```

Comment:
Make this part a function since its also used below for value_cache?

Reply:
Sure. Fixed.